abci/cmd/abci-cli/abci-cli.go

```diff
@@ -682,6 +700,38 @@ func cmdPrepareProposal(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func cmdProcessProposal(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
```

Comment:
If the app accepts empty blocks, would we want to do this?

Reply:
Both PrepareProposal and ProcessProposal in kvstore accept empty blocks (that is, requests with the field `Txs` having an empty array). PrepareProposal will just return an empty array in the field `TxRecords`, and ProcessProposal will return "accept". But empty blocks will never reach the application through `abci-cli`, because `abci-cli` does not accept an empty list of arguments for these two types of requests. So to keep it consistent with the application, probably the `if len(args) == 0 ...` block should be removed, for both commands.

Reply:
If it is correct behaviour for TM to accept empty blocks then Hernan is right, we should remove this check from both commands unless we think that for the kvstore app it does not make sense. I don't think it matters much, but if `ProcessProposal` accepts empty blocks, then `PrepareProposal` should too. We should keep the two consistent.

Reply:
So what is the correct behaviour? Does Tendermint accept empty blocks? And the applications?

Reply:
Ok, so to confirm, we need to remove the `if len(args) == 0 ...` block from PrepareProposal, ProcessProposal, FinalizeBlock, and CheckTx (not from Query)?

Reply:
I think so, although not sure about `Query`. Please don't modify `Query` for the time being

If you run into problems with regressions, etc, let me know

Reply:
I pushed a commit to allow empty blocks in PrepareProposal, ProcessProposal, and FinalizeBlock. CheckTx is not about blocks, it's only for one transaction, so I didn't change it.

Reply:
The changes look good but can you add a test in the files (the ones Sergio added) that have no transactions to make sure nothing will explode (maybe we have places where we assume that the transaction array is not empty) like in the actual function implementations within kvstore.

Reply:
Thanks Hernán. I just pushed a small commit to keep the tutorial in sync with `ex1.abci.out`